### PR TITLE
Pierre

### PR DESCRIPTION
**Pull request by ExxpresS**

### CODE REVIEW

File: Examcc/src/exo1/vectoriel.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package exo1;
+
+public class vectoriel {
+	double[] u;
+	double[] v;
+	
+	public static double [] crossProduct(double[] u, double[] v) throws Exception {
+		int size1 = u.length;
+		int size2 = v.length;
+		if (size1 != size2) throw new Exception("Cannot vectoriel cause list size different");
```
Comment (jpcosal) [inline Examcc/src/exo1/vectoriel.java:10]: Ok, this is one error case.
But to be more general, you just needed to check that both u.length=v.length=3

File: Examcc/src/exo2/SortedContainer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Arrays;
+
+public class SortedContainer implements ISortable {
+	
+	int[] element;
+	int listSize;
+	
+	public SortedContainer(int[] list){
+		element = list;
+		this.listSize = list.length;
```
Comment (jpcosal) [inline Examcc/src/exo2/SortedContainer.java:13]: Hmmm... The container is not sorted at the end of construction :(

File: Examcc/src/exo2/SortedContainer.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		return element;
+	}
+	public void add(int value) {
+		int [] newelem;
+		newelem=new int[(element.length)+1];
+		for (int i=0;i<(element.length); i++) {
+			newelem[i] = element[i];
+		}
+		newelem[newelem.length-1]=value;
+		sort(newelem,newelem.length);
```
Comment (jpcosal) [inline Examcc/src/exo2/SortedContainer.java:31]: Ok, this works but... Unfortunately this is totally ineffecient and resource consuming!
Each time you add an element, you make a copy of the previous list!
What if your initial list contains 1000000 elements?

File: Examcc/src/exo2/SortedContainerTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Test;
+
+class SortedContainerTest {
+
+	@Test
+	void testsort() {
+		int [] list = {3,2,6,5,4,9,7,8,1};
+		int [] res = {1,2,3,4,5,6,7,8,9};
+		SortedContainer C = new SortedContainer(list);
+		C.sort(list, 8);
```
Comment (jpcosal) [inline Examcc/src/exo2/SortedContainerTest.java:15]: Two comments:
- Should be done inside constructor...
- Why 8 when there are 9 elements?

File: Examcc/src/exo3/Dictionary.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package exo3;
+
+import java.util.ArrayList;
+
+public class Dictionary {
+	ArrayList<Word> al= new ArrayList<Word>();
```
Comment (jpcosal) [inline Examcc/src/exo3/Dictionary.java:6]: Use a map instead...

File: Examcc/src/exo3/Dictionary.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+	public void add (Word W) {
+		al.add(W);
+	}
+
+	public Word find (String id) {
+		for(Word word : al) {
```
Comment (jpcosal) [inline Examcc/src/exo3/Dictionary.java:13]: If you use a map you can speed up find and do it with **one single** instruction: `return al.get(id)`

File: Examcc/src/exo3/Word.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package exo3;
+
+public class Word extends Dictionary {
```
Comment (jpcosal) [inline Examcc/src/exo3/Word.java:3]: Why would `Word` extends `Dictionnary`??

**Review by jpcosal — COMMENTED** [review]
Lack of time?

**Comment by ExxpresS** [comment]
yes but I probably need to focus on exercise 4

**Pull request closed without merging**